### PR TITLE
fix(backend): On Pipeline delete, clean all associated Minio records

### PR DESCRIPTION
**Pull request by difince**
On Pipeline delete, clean all  Object Store (Minio) records associated with the Pipeline and its Versions

Fixes: https://github.com/kubeflow/pipelines/issues/7368

Signed-off-by: Diana Atanasova <dianaa@vmware.com>

**Description of your changes:**

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: backend/src/apiserver/resource/resource_manager.go [hunk 1]
```diff
@@ -233,20 +233,25 @@ func (r *ResourceManager) DeletePipeline(pipelineId string) error {
...
-	// either using async deletion in order for this method to be non-blocking
-	// or or exploring other performance optimization tools provided by gcs.
-	err = r.objectStore.DeleteFile(r.objectStore.GetPipelineKey(fmt.Sprint(pipelineId)))
-	if err != nil {
-		glog.Errorf("%v", errors.Wrapf(err, "Failed to delete pipeline file for pipeline %v", pipelineId))
+	errChan := r.objectStore.DeleteFiles(fileNames)
+	for err1 := range errChan {
+		glog.Errorf("%v", errors.Wrapf(err1, "Failed to delete pipeline file for pipeline %v.", pipelineId))
 		return nil
 	}
```
Comment (difince) [inline backend/src/apiserver/resource/resource_manager.go:253]: Note that I keep the current implementation - if Minios requests fail I just print the errors down and proceed with Pipeline deletion. 

> // Not fail the request if this step failed. A background run will do the cleanup.
	// https://github.com/kubeflow/pipelines/issues/388

I cannot find this background run that should cleanup the Minio records.

**Comment by difince** [comment]
assign @zijianjoy

**Comment by difince** [comment]
cc: @annajung, @juliusvonkohout 
If you have a change, please review

**Comment by juliusvonkohout** [comment]
Well it depends on the use case. I would like the pipeline definition AND pipeline artifacts be deleted from minio if they are deleted from the data base. Related is also https://github.com/kubeflow/pipelines/pull/7938
If @zijianjoy descides to move the pipeline definition to MySQL only then there are still artifacts from old runs.

**Comment by difince** [comment]
Closing the PR due to inactivity fortoo longtime

**Pull request closed without merging**